walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -28,28 +28,45 @@
     public int compare(Event a, Event b) {
       return Long.compare(a.getWhen().start(), b.getWhen().start());
     }};
-
-  public Collection<TimeRange> query(Collection<Event> eventCollection, MeetingRequest request) { 
+    
+  public Collection<TimeRange> query(Collection<Event> eventCollection, MeetingRequest request) {
+    Collection<String> attendees = new ArrayList<String>(); 
+    attendees.addAll(request.getOptionalAttendees());
+    attendees.addAll(request.getAttendees()); 
+    if (request.getOptionalAttendees().isEmpty()) {
+      attendees.removeAll(request.getOptionalAttendees());
+    }
+    if (request.getAttendees().isEmpty()) {
+      attendees.removeAll(request.getAttendees()); 
+    }
+    if (queryHelper(eventCollection, request, attendees).isEmpty()) {
+      attendees.removeAll(request.getOptionalAttendees());
+      return queryHelper(eventCollection, request, attendees);
+    }
+    return queryHelper(eventCollection, request, attendees);   
+  }
+  
+  private static Collection<TimeRange> queryHelper(Collection<Event> groupOfEvents, MeetingRequest request, Collection<String> attendees) {
     // Too long of a request
     if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
       return Collections.emptyList(); 
     }
     
-    // No events or attendees 
-    List<Event> events = new ArrayList<Event>(eventCollection);
-    List<String> meetingAttendees = new ArrayList<String>(request.getAttendees());
-    if (events.isEmpty() || meetingAttendees.isEmpty()) {
+    // No events 
+    List<Event> events = new ArrayList<Event>(groupOfEvents);
```

Comment:
Why are you doing this? Is it because you're accepting "groupOfEvents" as a "Collection"? Consider updating the function signature to accept a better collection type.